text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+To facilitate this, global injectors and extractors are optionally available. 
+However, there is no requirement to use this feature.
+
+**`GetExtractors() -> []http_extractor`**  
+To access the global extractor, the Propagation API provides a function which 
+returns an extractor.
+
+**`SetExtractors([]http_extractor)`**  
+To update the global extractor, the Propagation API provides a function which 
+takes an extractor.
+
+**`GetInjectors() -> []http_injector`**  
+To access the global injector, the Propagation API provides a function which 
+returns an injector.
+
+**`SetInjectors([]http_injector)`**  
+To update the global injector, the Propagation API provides a function which 
+takes an injector.
+
+# Prototypes
+
+**Erlang:** https://github.com/open-telemetry/opentelemetry-erlang-api/pull/4  
+**Go:** https://github.com/open-telemetry/opentelemetry-go/pull/381  
+**Java:** https://github.com/open-telemetry/opentelemetry-java/pull/655  
+**Python:** https://github.com/open-telemetry/opentelemetry-python/pull/278  
+**Ruby:** https://github.com/open-telemetry/opentelemetry-ruby/pull/147  
+
+# Examples
+
+It might be helpful to look at an example, written in pseudocode. Let's describe 
+a simple scenario, where `service A` responds to an HTTP request from a `client` 
+with the result of a request to `service B`.
+
+```
+client -> service A -> service B
+```
+
+Now, let's assume the `client` in the above system is version 1.0. With version 
+v2.0 of the `client`, `service A` must call `service C` instead of `service B` 
+in order to return the correct data.
+
+```
+client -> service A -> service C
+```
+
+In this example, we would like `service A` to decide on which backend service 
+to call, based on the client version. We would also like to trace the entire 
+system, in order to understand if requests to `service C` are slower or faster 
+than `service B`. What might `service A` look like?
+
+## Global initialization 
+First, during program initialization, `service A` configures correlation and tracing 
+propagation, and include them in the global list of injectors and extractors. 
+Let's assume this tracing system is configured to use B3, and has a specific 
+propagator for that format. Initializing the propagators might look like this:
+
+```php
+func InitializeOpentelemetry() {
+  // create the propagators for tracing and correlations.
+  bagExtract, bagInject = Correlations::HTTPPropagator()
+  traceExtract, traceInject = Tracer::B3Propagator()
+  
+  // add the propagators to the global list.
+  Propagation::SetExtractors(bagExtract, traceExtract)
+  Propagation::SetInjectors(bagInject, traceInject)
+}
+```
+
+## Extracting and injecting from HTTP headers
+These propagators can then be used in the request handler for `service A`. The 
+tracing and correlations concerns use the context object to handle state without 
+breaking the encapsulation of the functions they are embedded in.
+
+```php
```

Comment:
I'd second @carlosalberto - can you please post an example with the implicit context propagation and how APIs would look like

Reply:
This is added, in the section titled "Simplify the API with automated context propagation." 
https://github.com/tedsuo/rfcs/blob/context-prop-2/text/0066-separate-context-propagation.md#Simplify-the-API-with-automated-context-propagation